src/status_im/ui/screens/communities/views.cljs

```diff
@@ -0,0 +1,523 @@
+        :data                         (vals communities)
+        :render-fn                    (fn [community] [community-list-item community])}]]
+     (when config/communities-management-enabled?
+       [toolbar/toolbar
+        {:show-border? true
+         :center [quo/button {:on-press #(re-frame/dispatch [::create-pressed])}
+                  (i18n/label :t/create)]}])]))
+
+(fx/defn import-pressed
+  {:events [::import-pressed]}
+  [cofx]
+  (bottom-sheet/show-bottom-sheet cofx {:view :import-community}))
+
+(fx/defn create-pressed
+  {:events [::create-pressed]}
+  [cofx]
+  (bottom-sheet/show-bottom-sheet cofx {:view :create-community}))
+
+;; TODO: that's probably a better way to do this
+(defonce community-id (atom nil))
+
+(fx/defn invite-people-pressed
+  {:events [::invite-people-pressed]}
+  [cofx id]
+  (reset! community-id id)
+  (bottom-sheet/show-bottom-sheet cofx {:view :invite-people-community}))
+
+(fx/defn create-channel-pressed
+  {:events [::create-channel-pressed]}
+  [cofx id]
+  (reset! community-id id)
+  (bottom-sheet/show-bottom-sheet cofx {:view :create-community-channel}))
+
+(fx/defn community-created
+  {:events [::community-created]}
+  [cofx response]
+  (fx/merge cofx
+            (bottom-sheet/hide-bottom-sheet)
+            (communities/handle-response response)))
+
+(fx/defn community-imported
+  {:events [::community-imported]}
+  [cofx response]
+  (fx/merge cofx
+            (bottom-sheet/hide-bottom-sheet)
+            (communities/handle-response response)))
+
+(fx/defn people-invited
+  {:events [::people-invited]}
+  [cofx response]
+  (fx/merge cofx
+            (bottom-sheet/hide-bottom-sheet)
+            (communities/handle-response response)))
+
+(fx/defn community-channel-created
+  {:events [::community-channel-created]}
+  [cofx response]
+  (fx/merge cofx
+            (bottom-sheet/hide-bottom-sheet)
+            (communities/handle-response response)))
+
+(fx/defn import-confirmation-pressed
+  {:events [::import-confirmation-pressed]}
+  [cofx community-key]
+  (communities/import-community
+   cofx
+   community-key
+   #(re-frame/dispatch [::community-imported %])))
+
+(fx/defn create-confirmation-pressed
+  {:events [::create-confirmation-pressed]}
+  [cofx community-name community-description membership]
+  (communities/create
+   cofx
+   community-name
+   community-description
+   membership
+   ::community-created
+   ::failed-to-create-community))
+
+(fx/defn create-channel-confirmation-pressed
+  {:events [::create-channel-confirmation-pressed]}
+  [cofx community-channel-name community-channel-description]
+  (communities/create-channel
+   @community-id
+   community-channel-name
+   community-channel-description
+   ::community-channel-created
+   ::failed-to-create-community-channel))
+
+(fx/defn invite-people-confirmation-pressed
+  {:events [::invite-people-confirmation-pressed]}
+  [cofx user-pk]
+  (communities/invite-user
+   cofx
+   @community-id
```

Comment:
why atom here ?

Reply:
I would like to keep it locally, it's the kind of storage that I feel it should not be in the re-frame db, but local to the component.
Normally I would create a new instance of component initialization and keep it as local state, but given that is a `sheet` is a bit harder.
Do you think is best to keep it in app-db?

Reply:
i feel like events must be pure functions, and that's fine to have local atoms in components (views), but share an atom between events looks weird imo